test/unit/test_algorithms.cpp

```diff
@@ -9,10 +9,19 @@
 #include <arbor/util/compat.hpp>
 
 #include "algorithms.hpp"
-#include "threading/threading.hpp"
 #include "util/index_into.hpp"
 #include "util/meta.hpp"
 
+// (Pending abstraction of threading interface)
```

Comment:
How about indenting some of these includes? The pattern of "always align preprocessor to the left gutter" makes for ugly code in my opinion.

```
#include <arbor/version.hpp>
#if defined(ARB_TBB_ENABLED)
    #include "threading/tbb.hpp"
#elif defined(ARB_CTHREAD_ENABLED)
    #include "threading/cthread.hpp"
#else
    #include "threading/serial.hpp"
#endif
```

Reply:
I generally prefer not to indent directives unless it's really necessary for clarity, as the scoping rules for them are completely different from the scoping rules of the source code around them.

Reply:
It isn't about scoping, it is about making it easier to read and understand.
The additional clarity overides any potential scoping confusion for a proficient C++ developer.
There is an argument to be made, particularly at the top of a file like this, that there is a sort of  "preprocessor scoping".

Reply:
Consistency is also desirable, and having one convention for when it is just directives, and another when it is a mix of directives and code, is also unappealing. In short sequences like this I can't see it being a big gain in readability to indent them. It matters little in this instance, but I don't want to concede that indenting these guys is necessarily desirable.

Reply:
We discussed this with @vkarak 2 years ago when working on our coding standards.
After ~much bloodletting~ a discussion the decision was to indent preprocessor directives, though we never wrote it down.

Taking the original `src/threading/threading.hpp` file as a reference:
```
#if defined(ARB_HAVE_TBB)
    #include "tbb.hpp"
#elif defined(ARB_HAVE_CTHREAD)
    #include "cthread.hpp"
#else
    #define ARB_HAVE_SERIAL
    #include "serial.hpp"
#endif
```

Reply:
Pop it in the coding standards.

Reply:
We should copy those standards from the old `cell_algorithms` wiki over the the new wiki, where we might update them from time to time.